src/System.Drawing.Common/src/System/Drawing/Graphics.cs

```diff
@@ -20,6 +21,369 @@ public sealed partial class Graphics : MarshalByRefObject, IDisposable, IDeviceC
+
+                return dpi[0];
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the interpolation mode associated with this Graphics.
+        /// </summary>
+        public InterpolationMode InterpolationMode
+        {
+            get
+            {
+                int mode = 0;
+                int status = SafeNativeMethods.Gdip.GdipGetInterpolationMode(new HandleRef(this, NativeGraphics), out mode);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return (InterpolationMode)mode;
+            }
+            set
+            {
+                if (value < InterpolationMode.Invalid || value > InterpolationMode.HighQualityBicubic)
+                {
+                    throw new InvalidEnumArgumentException(nameof(value), unchecked((int)value), typeof(InterpolationMode));
+                }
+    
+                switch (value)
+                {
+                    case InterpolationMode.Default:
+                    case InterpolationMode.Low:
+                        value = InterpolationMode.Bilinear;
+                        break;
+                    case InterpolationMode.High:
+                        value = InterpolationMode.HighQualityBicubic;
+                        break;
+                    case InterpolationMode.Invalid:
+                        throw new ArgumentException(SR.GdiplusInvalidParameter);
+                    default:
+                        break;
+                }
+
+                int status = SafeNativeMethods.Gdip.GdipSetInterpolationMode(new HandleRef(this, NativeGraphics), unchecked((int)value));
+                SafeNativeMethods.Gdip.CheckStatus(status);
+            }
+        }
+
+        public bool IsClipEmpty
+        {
+            get
+            {
+                int isEmpty;
+                int status = SafeNativeMethods.Gdip.GdipIsClipEmpty(new HandleRef(this, NativeGraphics), out isEmpty);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return isEmpty != 0;
+            }
+        }
+
+        public bool IsVisibleClipEmpty
+        {
+            get
+            {
+                int isEmpty;
+                int status = SafeNativeMethods.Gdip.GdipIsVisibleClipEmpty(new HandleRef(this, NativeGraphics), out isEmpty);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return isEmpty != 0;
+            }
+        }
+
+        public float PageScale
+        {
+            get
+            {
+                var scale = new float[] { 0.0f };
+                int status = SafeNativeMethods.Gdip.GdipGetPageScale(new HandleRef(this, NativeGraphics), scale);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return scale[0];
+            }
+
+            set
+            {
+                if (value <= 0 || value > 1000000032)
+                {
+                    throw new ArgumentException(SR.GdiplusInvalidParameter);
+                }
```

Comment:
Ditto

Reply:
This fixes the test `PageScale_SetInvalid_ThrowsArgumentException` with libgdiplus